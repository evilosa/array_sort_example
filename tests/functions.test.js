const { guardNumber, calculateNumberWeight, sortingFunction } = require('../src/index')

describe('Function', () => {
  it('guardNumber should return the correct numbers weights', () => {
    expect(guardNumber('12345')).toBe('12345')
    expect(guardNumber('saa12dd45ddx2')).toBe('12452')
  })

  it('calculateNumberWeight should sort the incomming array by number weights', () => {
    expect(calculateNumberWeight('12345')).toBe(15)
    expect(calculateNumberWeight('saa12dd45ddx2')).toBe(14)
  })

  it('calculateNumberWeight should throw the TypeError for wrong type', () => {
    expect(() => { calculateNumberWeight(1234) }).toThrowErrorMatchingSnapshot()
  })

  it('sortingFunction should return the correct ordering indecies', () => {
    expect(sortingFunction('12345', '1234')).toEqual(1)
    expect(sortingFunction('1234', '12345')).toEqual(-1)
    expect(sortingFunction('11', '2000')).toEqual(-1)
    expect(sortingFunction('2000', '11')).toEqual(1)
    expect(sortingFunction('2002', '2002')).toEqual(0)
  })
})