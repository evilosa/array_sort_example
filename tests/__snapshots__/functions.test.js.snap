// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function calculateNumberWeight should throw the TypeError for wrong type 1`] = `"Calculate number weight: Wrong number 1234 was provided"`;
